### PR TITLE
Update snow consensus doc post X-chain linearization

### PR DESCRIPTION
**Pull request by exdx**
Updates the doc to reflect the recent changes in the X-chain that
changed the consensus protocol to Snowman from Avalanche. This was part
of the X-Chain linearization work.

Signed-off-by: Dan Sover <dan.sover@avalabs.org>

## Why this should be merged
Updates documentation to be accurate

## How this works
N/A

## How this was tested
N/A

### CODE REVIEW

**Review by yulin-dong — APPROVED** [review]

File: snow/README.md [hunk 1]
```diff
@@ -16,7 +16,7 @@ graph LR
...
+The Avalanche network consists of 3 built-in blockchains: X-Chain, C-Chain, and P-Chain. The X-Chain is used to manage assets and uses the Snowman consensus protocol. Previously the X-Chain used the Avalanche consensus protocol, which was DAG-based, but the consensus engine was updated to Snowman in avalanchego v1.10.1. The C-Chain is used to create and interact with smart contracts and uses the Snowman consensus protocol. The P-Chain is used to coordinate validators and stake and also uses the Snowman consensus protocol. At the time of writing, the Avalanche network has ~1200 validators. A set of validators makes up a subnet. Subnets can validate 1 or more chains. It is a common misconception that 1 subnet = 1 chain and this is shown by the primary subnet of Avalanche which is made up of the X-Chain, C-Chain, and P-Chain.
```
Comment (StephenButtolph) [inline snow/README.md:19]: > was updated to Snowman in avalanchego v1.10.1.

Should be: "was updated to Snowman in the Cortina network upgrade."

Also I feel like we could more susinctly say that all chains are now based on Snowman consensus.. Rather than explicitly saying that the P-chain and the C-chain use Snowman

Comment (exdx) [reply snow/README.md:19]: Sure, sounds good. I agree it's overly verbose as-is.

**Review by StephenButtolph — APPROVED** [review]

File: snow/README.md [hunk 2]
```diff
@@ -16,7 +16,9 @@ graph LR
 
 ## Intro
 
-The Avalanche network consists of 3 built-in blockchains: X-Chain, C-Chain, and P-Chain. The X-Chain is used to manage assets and uses the Avalanche consensus protocol. The C-Chain is used to create and interact with smart contracts and uses the Snowman consensus protocol. The P-Chain is used to coordinate validators and stake and also uses the Snowman consensus protocol. At the time of writing, the Avalanche network has ~1200 validators. A set of validators makes up a subnet. Subnets can validate 1 or more chains. It is a common misconception that 1 subnet = 1 chain and this is shown by the primary subnet of Avalanche which is made up of the X-Chain, C-Chain, and P-Chain.
+The Avalanche network consists of 3 built-in blockchains: X-Chain, C-Chain, and P-Chain. All three chains rely on the Snowman Consensus protocol. Previously the X-Chain used the Avalanche consensus protocol, which was DAG-based, but the consensus engine was updated to Snowman in the Cortina network update.
```
Comment (hexfusion) [inline snow/README.md:19]: nit: This still feels a bit wordy maybe we can compress this a little further?
```suggestion
The Avalanche network consists of 3 built-in blockchains: X-Chain, C-Chain, and P-Chain. All three chains rely on the Snowman consensus protocol. The X-Chain which previously used Avalanche consensus (DAG-based) was upgraded to Snowman in the Cortina network update.

**Review by hexfusion — APPROVED** [review]

File: snow/README.md (hunk 2 above)
Comment (exdx) [reply snow/README.md:19]: Updated:
```
The Avalanche primary network consists of 3 built-in blockchains: the X-Chain, C-Chain, and P-Chain. All three chains rely on the Snowman consensus protocol. The X-Chain, which previously used DAG-based Avalanche consensus, was upgraded to Snowman in the Cortina network update. 
```

File: snow/README.md [hunk 3]
```diff
@@ -16,7 +16,9 @@ graph LR
...
+The Avalanche primary network consists of 3 built-in blockchains: the X-Chain, C-Chain, and P-Chain. All three chains rely on the Snowman consensus protocol. The X-Chain, which previously used DAG-based Avalanche consensus, was upgraded to Snowman in the Cortina network update. 
+
+The X-Chain is used to manage assets. The C-Chain is used to create and interact with smart contracts. The P-Chain is used to coordinate validators and stake. At the time of writing, the Avalanche network has ~1200 validators. A set of validators makes up a subnet. Subnets can validate 1 or more chains. It is a common misconception that 1 subnet = 1 chain and this is shown by the primary subnet of Avalanche which is made up of the X-Chain, C-Chain, and P-Chain.
```
Comment (abi87) [inline snow/README.md:21]: Should we keep a structure more similar to the original for the X-chain? Something like
`All three chains rely on the Snowman consensus protocol.
The X-Chain is used to manage assets; it previously used DAG-based Avalanche consensus and was upgraded to Snowman in the Cortina network update.`

**Review by abi87 — CHANGES_REQUESTED** [review]

File: snow/README.md (hunk 3 above)
Comment (exdx) [reply snow/README.md:21]: Thanks @abi87. I'm not sure I can go either way. I can't merge regardless, so let's see how others feel.

Comment (dhrubabasu) [reply snow/README.md:21]: My impression of this doc is the "current state of affairs" - for example, we don't mention Snowman++ here
